tooi/api/statuses.py

```diff
@@ -41,11 +42,44 @@ async def post(
     return await request("POST", "/api/v1/statuses", headers=headers, json=payload)
 
 
+async def edit(
+    status_id: str,
+    status: str,
+    visibility: Visibility = Visibility.Public,
+    sensitive: bool = False,
+    spoiler_text: str | None = None,
+) -> Response:
+    """
+    Edit an existing status.
+    https://docs.joinmastodon.org/methods/statuses/#edit
+    """
+
+    payload = drop_empty_values({
+        "status": status,
+        "visibility": visibility,
+        "sensitive": sensitive,
+        "spoiler_text": spoiler_text,
+    })
+
+    return await request("PUT", f"/api/v1/statuses/{status_id}", json=payload)
+
+
 def drop_empty_values(data: Dict[Any, Any]) -> Dict[Any, Any]:
     """Remove keys whose values are null"""
     return {k: v for k, v in data.items() if v is not None}
 
 
+async def get_status_source(status_id: str):
+    """
+    Fetch the original plaintext source for a status.  Only works on locally-posted statuses.
+    https://docs.joinmastodon.org/methods/statuses/#source
+    """
+    path = f"/api/v1/statuses/{status_id}/source"
+    response = await request("GET", path)
+    json = response.json()
+    return from_dict(StatusSource, json)
```

Comment:
I'd like to keep the api module functions unaware of entities and returning just the response, for consistency.

In toot i started with all api functions returning the decoded data, but then ran into situations where i don't want the data decoded (e.g. want to print out the json to screen), or needing data from the response headers.

We could add some "high level" api functions in the `data` module, although I don't think it's warranted here.